hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsMesosConstraints.java

```diff
@@ -0,0 +1,104 @@
+package org.apache.mesos.hdfs.scheduler;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.mesos.Protos.Attribute;
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.Value.Range;
+
+/**
+ * HDFS Mesos offer constraints checker class implementation.
+ */
+public class HdfsMesosConstraints {
+
+  private final Map<String, String> constraints;
+
+  public HdfsMesosConstraints(Map<String, String> mesosSlaveConstraints) {
+    this.constraints = mesosSlaveConstraints;
+  }
+
+  public boolean constraintsAllow(Offer offer) {
+    List<Attribute> attributes = offer.getAttributesList();
+
+    for (Map.Entry<String, String> constraintEntry : this.constraints
+        .entrySet()) {
+      boolean found = false;
+      String constraintName = constraintEntry.getKey();
+      String constraintValue = constraintEntry.getValue();
+      for (Attribute attribute : attributes) {
+
+        if (attribute.getName().equals(constraintName)) {
+
+          switch (attribute.getType()) {
+          case RANGES:
+            if (attribute.hasRanges()) {
+              try {
+                Long value = Long.parseLong(constraintValue);
+                for (Range r : attribute.getRanges().getRangeList()) {
+                  if ((!r.hasBegin() || value >= r.getBegin())
+                      && (!r.hasEnd() || value <= r.getEnd())) {
+                    found = true;
+                    break;
+                  }
+                }
+              } catch (NumberFormatException e) {
+                found = false;
+                // not a range attribute
+              }
+            }
+            break;
+          case SCALAR:
+            if (attribute.hasScalar()) {
+              try {
+                if (attribute.getScalar().getValue() >= Double
+                    .parseDouble(constraintValue)) {
+                  found = true;
+                }
+              } catch (NumberFormatException e) {
+                found = false;
+                // not a scalar attribute
```

Comment:
It not an error, it is ok that constraint value is not of type that range expects. It just mean this doesn't match